rtsf-at-checkout-device-scale/driver/scaleDriver.go

```diff
@@ -86,7 +87,7 @@ func (drv *ScaleDriver) HandleReadCommands(deviceName string, protocols map[stri
 	res = make([]*dsModels.CommandValue, len(reqs))
 
 	if !drv.scaleConnected {
-		return nil, nil
+		return nil, errors.New("scale is not connected")
```

Comment:
will this change be ok for the original requirement?

Reply:
do we have simulator to simulate this case?  if so, we probably could confirm it

Reply:
@brian-intel and I have discussed about this, this seems to be reasonable change, if simulator doesn't require to bring the whole stack to verify this case, then I can do that now, the whole stack to be ready will require all services upgraded to EdgeX2.2, not sure if we are there yet.

Reply:
yeah, i understand that but my main concern is about the original requirement: if scale device is not connected, will the whole app crashed due to no connection and no readings then?